packages/ag-charts-community/src/scene/layer.ts

```diff
@@ -0,0 +1,201 @@
+import { BBox } from './bbox';
+import type { HdpiCanvas } from './canvas/hdpiCanvas';
+import { nodeCount } from './debug.util';
+import { Group } from './group';
+import type { LayersManager, ZIndexSubOrder } from './layersManager';
+import { type ChildNodeCounts, Node, RedrawType, type RenderContext } from './node';
+import { Translatable } from './transformable';
+
+export class Layer extends Group {
+    static override className = 'Layer';
+
+    static override is(value: unknown): value is Layer {
+        return value instanceof Layer;
+    }
+
+    private layer?: HdpiCanvas;
+    private lastBBox?: BBox = undefined;
+
+    constructor(
+        protected override readonly opts?: {
+            name?: string;
+            zIndex?: number;
+            zIndexSubOrder?: ZIndexSubOrder;
+            deriveZIndexFromChildren?: boolean; // TODO remove feature
+        }
+    ) {
+        super(opts);
+    }
+
+    override markDirty(type = RedrawType.TRIVIAL) {
+        super.markDirty(type, this.layer != null ? RedrawType.TRIVIAL : type);
```

Comment:
Is the logic here correct? `this.layer` being `undefined` is just because of lazy initialisation? If so, I would have thought that `markDirty()` should always percolate dirty status of `TRIVIAL`?

Reply:
That's the current logic, I'll change it